### PR TITLE
HDDS-3191: switch from SCMPipelineManager to PipelineManagerV2Impl

### PR DESCRIPTION
**Pull request by GlenGeng**
## What changes were proposed in this pull request?

switch from SCMPipelineManager to PipelineManagerV2Impl.
mainly focus on fixing unit test

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-3191

## How was this patch tested?

unit tests

### CODE REVIEW

**Comment by GlenGeng** [comment]
Hello @elek @adoroszlai  Would you please help check a GitHub CI problem ? 

I want to push 2 commit from my dev branch `GlenGeng:HDDS-3191` to branch `apache:HDDS-2823`, meanwhile I also tried CI in my personal repo (submit a PR from branch `GlenGeng:HDDS-3191` to branch `GlenGeng:HDDS-2823`, see https://github.com/GlenGeng/hadoop-ozone/pull/2). 

It is pretty sure that branch `apache:HDDS-2823` is the same as branch `GlenGeng:HDDS-2823`. I have succeed in my [person repo](https://github.com/GlenGeng/hadoop-ozone/pull/2/checks), but failed several time in [apache repo](https://github.com/apache/hadoop-ozone/pull/1151/checks?check_run_id=829732353).

Seems these failed cases are related to raft leader election, which has randomness. Would you please help have a look ?

Thanks!

**Comment by GlenGeng** [comment]
also see this flaky case failure in `master` branch

see https://github.com/apache/hadoop-ozone/pull/1147

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMHAConfiguration.java [hunk 1]
```diff
@@ -78,7 +78,7 @@
       description = "The size of the raft segment used by Apache Ratis on" +
           " SCM. (16 KB by default)"
   )
-  private long raftSegmentSize = 16L * 1024L;
+  private double raftSegmentSize = 16L * 1024L;
```
Comment (xiaoyuyao) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMHAConfiguration.java:81]: Is there a reason to change long to double here and force conversion later?

File: hadoop-ozone/recon/pom.xml [hunk 2]
```diff
@@ -108,6 +108,7 @@
             </goals>
             <configuration>
               <arguments>pnpm config set store-dir ~/.pnpm-store</arguments>
+              <npmInheritsProxyConfigFromMaven>false</npmInheritsProxyConfigFromMaven>
```
Comment (xiaoyuyao) [inline hadoop-ozone/recon/pom.xml:111]: This change seems unrelated. Can you merge it from master?

**Comment by timmylicheng** [comment]
LGTM, +1.
See if failed tests can be cleared.

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMHAConfiguration.java (hunk 1 above)
Comment (xiaoyuyao) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMHAConfiguration.java:81]: I see. Thanks for the details.

**Comment by xiaoyuyao** [comment]
+1, I will merge it shortly.
